### ISSUE TITLE
yoda rule exceptRange false positives

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#12883]
**Tell us about your environment**

* **ESLint Version:** 7.0.0-alpha.0
* **Node Version:** v12.14.0
* **npm Version:** v6.13.4

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    parserOptions: {
        ecmaVersion: 2015
    },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

These are two different issues.

1. Possible inconsistency between `"never"` and `"always"`.

[Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IHlvZGE6IFtcImVycm9yXCIsIFwibmV2ZXJcIiwgeyBleGNlcHRSYW5nZTogdHJ1ZSB9XSAqL1xuXG5pZiAoMCA8IHggJiYgeCA8IHNvbWV0aGluZykge30gLy8gbm8gZXJyb3IiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

```js
/* eslint yoda: ["error", "never", { exceptRange: true }] */

if (0 < x && x < something) {} // no error
```

[Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IHlvZGE6IFtcImVycm9yXCIsIFwiYWx3YXlzXCIsIHsgZXhjZXB0UmFuZ2U6IHRydWUgfV0gKi9cblxuaWYgKHNvbWV0aGluZyA8IHggJiYgeCA8IDEwMCApIHt9IC8vIGVycm9yIG9uIHggPCAxMDAiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

```js
/* eslint yoda: ["error", "always", { exceptRange: true }] */

if (something < x && x < 100 ) {} // error on x < 100
```

2. Possible inconsistency between how the rule treats numbers and strings.

[Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IHlvZGE6IFtcImVycm9yXCIsIFwibmV2ZXJcIiwgeyBleGNlcHRSYW5nZTogdHJ1ZSB9XSAqL1xuXG5pZiAoMSA8IHggJiYgeCA8IE1BWCApIHt9IC8vIG5vIGVycm9yXG5cbmlmIChcImFcIiA8IHggJiYgeCA8IE1BWCApIHt9IC8vIGVycm9yIG9uIFwiYVwiIDwgeCIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

```js
/* eslint yoda: ["error", "never", { exceptRange: true }] */

if (1 < x && x < MAX ) {} // no error

if ("a" < x && x < MAX ) {} // error on "a" < x
```

```bash
eslint index.js
```

**What did you expect to happen?**

Maybe no errors.

**What actually happened? Please include the actual, raw output from ESLint.**

Two errors noted in the comments.

**Are you willing to submit a pull request to fix this bug?**

Yes.

**Comment by kaicataldo** [comment]
These both looks like bugs to me :+1:

Can we split this into two issues (or at least implement them in separate PRs)?

**Comment by anikethsaha** [comment]
@mdjermanovic 

for this example 
```js
/* eslint yoda: ["error", "always", { exceptRange: true }] */

if (something < x && x < 100 ) {} // error on x < 100
```

`exceptRange` options is for `never` so when testing it with `never` its lint free

[demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IHlvZGE6IFtcImVycm9yXCIsIFwibmV2ZXJcIiwgeyBleGNlcHRSYW5nZTogdHJ1ZSB9XSAqL1xuXG5pZiAoc29tZXRoaW5nIDwgeCAmJiB4IDwgMTAwICkge30gLy8gZXJyb3Igb24geCA8IDEwMCIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

**Comment by mdjermanovic** [comment]
> `exceptRange` options is for `never` right

That's indeed what documentation says:

> The default "never" option can have exception options in an object literal

However, that text was added later in #5596. 

By the original issue #1561 and PR #1570 it doesn't seem that the `exceptRange` option was meant to apply to `"never"` only.

Also, the same sentence in documentation is incorrect for the `"onlyEquality"` option, because it does apply to `"always"` as well:

```js
/*eslint yoda: ["error", "always", { "onlyEquality": true }]*/

if (x < 1) {} // no error
```

**Comment by anikethsaha** [comment]
thanks for the info. 
 👍 

I will try to fix the 2nd issue (`2. Possible inconsistency between how the rule treats numbers and strings.` ) through #13052
